base/acme/src/main/java/org/dogtagpki/acme/server/ACMEEngine.java

```diff
@@ -746,4 +748,52 @@ public void parseCSRSAN(SubjectAlternativeNameExtension sanExt, Set<String> dnsN
             }
         }
     }
+
+    public void validateRevocation(ACMEAccount account, ACMERevocation revocation) throws Exception {
+
+        // RFC 8555 Section 7.6: Certificate Revocation
+        //
+        // The server MUST consider at least the following accounts authorized
+        // for a given certificate:
+        // -  the account that issued the certificate.
+        // -  an account that holds authorizations for all of the identifiers in
+        //    the certificate.
+
+        // Case 1: validate using order information (if it still exists)
+
+        String certBase64 = revocation.getCertificate();
+        byte[] certBytes = Utils.base64decode(certBase64);
+        X509CertImpl cert = new X509CertImpl(certBytes);
+
+        BigInteger serialNumber = cert.getSerialNumber();
+        logger.info("Serial Number: 0x" + serialNumber.toString(16));
+
+        String certID = Base64.encodeBase64URLSafeString(serialNumber.toByteArray());
+
+        logger.info("Finding order that issued the certificate");
+        ACMEOrder order = database.getOrderByCertificate(certID);
+
+        if (order != null) {
+            logger.info("Order ID: " + order.getID());
+
+            // check order ownership
+            if (!order.getAccountID().equals(account.getID())) {
+                // TODO: generate proper exception
+                throw new Exception("Account did not issue the certificate");
+            }
+
+            // No need to check order status since it's guaranteed to be valid.
+            // No need to check order expiration since it's irrelevant for revocation.
+
+            logger.info("Account authorized to revoke certificate");
+            return;
+        }
+
+        logger.info("Order not found");
+
+        // TODO: Case 2: validate using cert identifiers authorizations
```

Comment:
For this, perhaps we could a method to `ACMEDatabase`, e.g.:

```java
Set<String> /*account ids*/ getAuthorizedAccountsForIdentifier(ACMEIdentifier id);
```

The engine can then take the intersection of all the sets.

RFC 8555 is a little vague but I would consider current (i.e. non-expired and valid) authorizations only.

Reply:
In #348 I added `getAuthorizationsByAccount()` instead since it seems to be easier to translate into actual database query compared to `getAuthorizedAccountsForIdentifier()` which may involve wildcards.